diff --git a/app/controllers/microposts_controller.rb b/app/controllers/microposts_controller.rb
index f7caba8..3cafe5f 100644
--- a/app/controllers/microposts_controller.rb
+++ b/app/controllers/microposts_controller.rb
@@ -3,11 +3,6 @@ class MicropostsController < ApplicationController
   before_action :logged_in_user, only: [:create, :destroy]
   before_action :correct_user,   only: :destroy
 
-  def index
-    @microposts = current_user.microposts
-    render json: @microposts
-  end
-
   def create
     @micropost = current_user.microposts.build(micropost_params)
     @micropost.image.attach(params[:micropost][:image])
