diff --git a/config/environments/test.rb b/config/environments/test.rb
index 88ed2ba..a8d5147 100644
--- a/config/environments/test.rb
+++ b/config/environments/test.rb
@@ -13,6 +13,8 @@ Rails.application.configure do
   # preloads Rails for running tests, you may have to set it to true.
   config.eager_load = false
 
+  config.log_level = :warn
+
   # Configure public file server for tests with Cache-Control for performance.
   config.public_file_server.enabled = true
   config.public_file_server.headers = {
