diff --git a/app/models/user.rb b/app/models/user.rb
index 8c6161c..94989df 100644
--- a/app/models/user.rb
+++ b/app/models/user.rb
@@ -104,7 +104,7 @@ class User < ApplicationRecord
                      WHERE  follower_id = :user_id"
     Micropost.where("user_id IN (#{following_ids})
                      OR user_id = :user_id", user_id: id)
-                   # .includes(:user, image_attachment: :blob)
+             .includes(:user, image_attachment: :blob)
   end
 
   # Follows a user.
